sqlmesh/core/renderer.py

```diff
@@ -61,6 +61,7 @@ def __init__(
         jinja_macro_registry: t.Optional[JinjaMacroRegistry] = None,
         python_env: t.Optional[t.Dict[str, Executable]] = None,
         only_latest: bool = False,
+        resolve_tables: bool = True,
```

Comment:
Can we push resolution of tables to later all the time? I'm wondering if that will simplify the code (assuming its possible)

Reply:
Not sure I understand what you're suggesting exactly

Reply:
My understanding is this boolean denotes whether we should resolve tables here or not. I'm assuming that means we resolve them later? And if so, why wouldn't we always do it there.

Reply:
Actually, let me refactor

Reply:
refactored, please take a look